platform/pulp/platform/fields.py

```diff
@@ -0,0 +1,96 @@
+"""
+Custom django fields provided by pulp platform
+"""
+import json
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import six, ugettext_lazy as _
+
+
+def _parse_json(json_string):
+    """
+    Parse a json string
+
+    :param json_string: A json string to parse
+    :type json_string: str
+
+    :return: A python object built from the json string
+    :rtype: Any
+    """
+    try:
+        return json.loads(json_string)
+    except Exception:
+        raise ValidationError(_("Invalid JSON stored in database"))
+
+
+class JSONField(models.TextField):
+    """
+    A custom django field to serialize dicts into a text field
+
+    """
+    def from_db_value(self, value, expression, connection, context):
+        """
+        Converts a value as returned by the database to a Python object
+
+        :param value: The value to convert to Python
+        :type value: Any
+
+        :param expression:
+        :param connection:
+        :param context:
```

Comment:
Per [the docs](https://docs.djangoproject.com/en/1.8/ref/models/fields/#django.db.models.Field.from_db_value)

> For performance reasons, from_db_value is not implemented as a no-op on fields which do not require it (all Django fields). Consequently you may not call super in your definition.